01 SFC Hello VueJS/README.md

```diff
@@ -0,0 +1,264 @@
+
+# Introduction
+
+In this sample we are going to create our first Vue.js SFC and connect it with the DOM.
+
+We will take a startup point sample _00 SFC Boilerplate_.
+
+Summary steps:
+
+- Install Vue.js devtools.
+- Install `vue.js` library and others dependencies.
+- Configure webpack to work with `vue.js`.
+- Update `index.html`.
+- Update `main.ts`.
+
+# Steps to build it
+
+## Prerequisites
+
+You will need to have Node.js installed in your computer. In order to follow this step guides you will also need to take sample _00 SFC BoilerPlate_ as a starting point.
+
+## Steps
+
+- `npm install` to install previous sample dependencies:
+
+```
+npm install
+```
+
+- To get a better experience debugging `vue` we can install [Vue devtools](https://github.com/vuejs/vue-devtools). It's a Chrome extension.
+
+- Install `vue` library as project dependency:
+
+```
+npm install vue --save
+```
+
+- To use SFC there install vue-loader and vue-template-compiler:
+
+```
+npm install vue-loader vue-template-compiler --save
+```
+
+- This library has its own TypeScript definition file, so we don't need install other typings.
+
+- Configure `webpack.config` to set `vue.js` as vendor:
+
+### ./webpack.config.js
+
+```diff
+...
+  entry: {
+    app: './main.ts',
++   vendor: [
++     'vue',
++   ],
+    vendorStyles: [
+      '../node_modules/bootstrap/dist/css/bootstrap.css',
+    ],
+  },
+...
+```
+
+- Recommended configuration for [`tsconfig.json`](https://vuejs.org/v2/guide/typescript.html#Recommended-Configuration):
+
+### ./tsconfig.json
+```diff
+{
+  "compilerOptions": {
+    "target": "es6",
+    "module": "es6",
+    "moduleResolution": "node",
+    "declaration": false,
+    "noImplicitAny": false,
+    "sourceMap": true,
+    "noLib": false,
+-   "suppressImplicitAnyIndexErrors": true
++   "suppressImplicitAnyIndexErrors": true,
++   "strict": true
+  },
+  "compileOnSave": false,
+  "exclude": [
+    "node_modules"
+  ]
+}
+
+```
+
+- There are [two builds available](https://vuejs.org/v2/guide/installation.html#Explanation-of-Different-Builds) to work with `Vue.js`.
+
+  - `Runtime + Compiler`: needed to compile templates on the client side, e.g. passing a string to `template` property.
+  - `Runtime-only`: when using a `pre-compiler` like `vue-loader`, `vueify`, etc.
+
+- For now, we could start using `first one` to keep this sample as simple as possible. To configure it for webpack:
+
+### ./webpack.config.js
+
+```diff
+...
+resolve: {
+-    extensions: ['.js', '.ts'],
++    extensions: ['.js', '.ts', 'vue'],
++    alias: {
++      'vue$': 'vue/dist/vue.esm.js'
++    },
+  },
+...
+  module: {
+    rules: [
++     {
++       test: /\.vue$/,
++       exclude: /node_modules/,
++       loader: 'vue-loader'
++     },
+      {
+        test: /\.ts$/,
+        exclude: /node_modules/,
+        use: {
+          loader: 'ts-loader',
+          options: {
++           appendTsSuffixTo: [/\.vue$/],
```

Comment:
This is the key to use `ts-loader` instead of `awesome-typescript-loader` isnt it?

Reply:
Right! I think the awesome-typescript-loader team is working on the integration.